src/legacy/ui/public/vis/editors/default/components/default_editor_agg_params_helper.test.ts

```diff
@@ -0,0 +1,210 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { AggConfig, VisState } from 'ui/vis';
+import { FieldParamType, AggType } from 'ui/agg_types';
+import { IndexPattern } from 'ui/index_patterns';
+import {
+  getAggParamsToRender,
+  getError,
+  getAggTypeOptions,
+  isInvalidParamsTouched,
+} from './default_editor_agg_params_helper';
+
+jest.mock('ui/agg_types', () => ({
+  aggTypes: {
+    byType: {
+      metrics: [],
+      buckets: [],
+    },
+  },
+}));
+jest.mock('../default_editor_utils', () => ({
+  groupAggregationsBy: jest.fn(() => ['indexedFields']),
+}));
+
+describe('DefaultEditorAggParams helpers', () => {
+  describe('getAggParamsToRender', () => {
+    let agg: AggConfig = {
+      type: {
+        params: [{ name: 'interval' }],
+      },
+    };
+    const editorConfig = {
+      interval: {
+        hidden: true,
+      },
+    };
+    const state = {} as VisState;
+    const responseValueAggs = null;
+
+    it('should not create any param if there is no agg type, params do not have editorComponents or a parma is hidden', () => {
```

Comment:
Let's separate these case into 3 `it`s. This case describes 3 options but tests only one.

```suggestion
    it('should not create any param if there is no agg type, params do not have editorComponents or a param is hidden', () => {
```